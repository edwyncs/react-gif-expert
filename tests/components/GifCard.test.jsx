const { render, screen } = require("@testing-library/react");
const { GifCard } = require("../../src/components/GifCard");

describe('Pruebas del componente GifItem', () => { 
    const title = 'titulo';
    const url = 'http://www.one-punch.com/one.gif';
    
    test('debe hacer match con el snapshot', () => {
        const {container} = render (<GifCard title={title} url={url}></GifCard>);
        expect(container).toMatchSnapshot();
    });

    test('debe de mostrar la imagen con el URL y el ALT indicado', () => { 
        render (<GifCard title={title} url={url}></GifCard>);
        //screen.debug();   //muestra el codigo HTML
        //expect(screen.getByRole('img').src).toBe(url)     //para comparar el url del html con el snapshot
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(title).toBe(alt);
    });
    
    test('debe de mostrar el titulo en el componente', () => { 
        render (<GifCard title={title} url={url}></GifCard>);
        expect(screen.getByText(title)).toBeTruthy();
     })
 });