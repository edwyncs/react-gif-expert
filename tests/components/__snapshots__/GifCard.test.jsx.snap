// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente GifItem debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="titulo"
      src="http://www.one-punch.com/one.gif"
    />
    <p>
      titulo
    </p>
  </div>
</div>
`;
